src/Sema.zig

```diff
@@ -30702,7 +30702,8 @@ fn addReferencedBy(
     src: LazySrcLoc,
     decl_index: Decl.Index,
 ) !void {
-    if (sema.mod.comp.reference_trace == @as(u32, 0)) return;
+    if (sema.mod.comp.reference_trace == 0) return;
+    if (src == .unneeded) return; // we can't use NeededSourceLocation since sites handling that assume it means a compile error
```

Comment:
Shouldn't this be an assert? If you're paranoid about bugs, how about an assert in debug mode?

Reply:
Is this what you're suggesting?
```zig
if (src == .unneeded) {
    if (std.debug.runtime_safety) unreachable else return;
}
```
That'd be fine with me - it prevents blatant crashes in release compilers, but allows us to notice the issue in debug builds. Happy to make that change if you want.
Per my previous comment, I think this is a good example of why `NeededSourceLocation` is a questionable system, and indicative that we should just have more `LazySrcLoc` fields for the stuff we currently use `NeededSourceLocation` for.

Reply:
Yeah that's what I'm suggesting. I'm in favor of moving away from .unneeded and towards having more LazySrcLoc fields :+1: